src/libraries/Microsoft.PowerFx.Core/Types/BuiltInLazyTypes/AttachmentType.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Types.BuiltInLazyTypes
+{
+    // Type represents attachment type which are delay loaded fields at runtime
+    internal class AttachmentType : RecordType
```

Comment:
>RecordType [](http://example.com/codeflow?start=36&length=10)

Is it a record or a table type? On DTypeTests.cs (DTypeToStringRepresentation) it is listed as `r*` (lazy table), but here it is listed as a record.

Reply:
Added assert. The wrapped type is always a record, however, it may be a lazy table if the attachment type was originally a table.

Reply:
Updated tests to capture both wrapped tables and records.